src/build.d

```diff
@@ -262,121 +262,87 @@ alias backend = makeDep!((builder, dep) => builder
-                }
-                catch (ProcessException)
-                {
-                    // git not installed
-                }
+                auto gitResult = ["git", "describe", "--dirty"].run;
+                if (gitResult.status == 0)
+                    ver = gitResult.output.strip;
             }
-            // version fallback
-            if (ver.length == 0)
-                ver = dmdRepo.buildPath("VERSION").readText;
-            updateIfChanged(target, ver);
-        };
-    }
-    return new DependencyRef(dep);
-});
+            catch (ProcessException)
+            {
+                // git not installed
+            }
+        }
+        // version fallback
+        if (ver.length == 0)
+            ver = dmdRepo.buildPath("VERSION").readText;
+        updateIfChanged(dep.target, ver);
+    })
+);
 
-alias sysconfDirFile = memoize!(function()
-{
-    Dependency dep;
-    with(dep)
-    {
-        msg = "(TX) SYSCONFDIR";
-        sources = [thisBuildScript];
-        target = env["G"].buildPath("SYSCONFDIR.imp");
-        commandFunction = ()
-        {
-            updateIfChanged(target, env["SYSCONFDIR"]);
-        };
-    }
-    return new DependencyRef(dep);
-});
+alias sysconfDirFile = makeDep!((builder, dep) => builder
+    .msg("(TX) SYSCONFDIR")
+    .target(env["G"].buildPath("SYSCONFDIR.imp"))
+    .commandFunction(() {
+        updateIfChanged(dep.target, env["SYSCONFDIR"]);
+    })
+);
 
 /**
 Dependency for the DMD executable.
 
 Params:
   extra_flags = Flags to apply to the main build but not the dependencies
 */
-alias dmdExe = memoize!(function(string targetSuffix, string[] extraFlags...)
-{
+alias dmdExe = makeDepWithArgs!((MethodInitializer!DependencyRef* builder, DependencyRef dep, string targetSuffix, string[] extraFlags) {
     const dmdSources = sources.dmd.chain(sources.root).array;
 
-    // Main DMD build dependency
-    Dependency dep;
-    with (dep)
-    {
+    string[] platformArgs;
+    version (Windows)
+        platformArgs = ["-L/STACK:8388608"];
+
+    builder
         // newdelete.o + lexer.a + backend.a
-        sources = dmdSources.chain(lexer.targets, backend.targets).array;
-        target = env["DMD_PATH"] ~ targetSuffix;
-        msg = "(DC) DMD%s %-(%s, %)".format(targetSuffix, dmdSources.map!(e => e.baseName).array);
-        deps = [versionFile, sysconfDirFile, lexer, backend];
-        string[] platformArgs;
-        version (Windows)
-            platformArgs = ["-L/STACK:8388608"];
-        command = [
+        .sources(dmdSources.chain(lexer.targets, backend.targets).array)
+        .target(env["DMD_PATH"] ~ targetSuffix)
+        .msg("(DC) DMD%s %-(%s, %)".format(targetSuffix, dmdSources.map!(e => e.baseName).array))
+        .deps([versionFile, sysconfDirFile, lexer, backend])
+        .command([
             env["HOST_DMD_RUN"],
-            "-of" ~ target,
+            "-of" ~ dep.target,
             "-vtls",
-            "-J"~env["G"],
```

Comment:
As far as I can see, in order to use the output of `versionFile` target, we need to add the `"-J"~env["G"]` string import. Am I missing something?

Reply:
Its part of `flags["DFLAGS"]`, see

https://github.com/marler8997/dmd/blob/6082657531b5fa5da5e91c78d8b4a88afcfe403d/src/build.d#L798

Reply:
Thanks, indeed I missed that part.